alg/gdal_alg_priv.h

```diff
@@ -206,6 +207,8 @@ bool GDALTransformIsTranslationOnPixelBoundaries(
 bool GDALTransformIsAffineNoRotation(GDALTransformerFunc pfnTransformer,
                                      void *pTransformerArg);
 
+bool GDALTransformHasFastClone(void *pTransformerArg);
```

Comment:
is there a chance we do not use `void *` anymore? 
They make the code almost impossible to read.

Reply:
Without impacting too much the API, we could probably have a "typedef void* GDALTransformerArg:". Doing better would involve ditching those C based APIs and just use C++ type inheritance, but that would impact external code (there are users having custom transformer implementations)

Reply:
I wonder if it's feasible to move to C++ type inheritance internally but have a compatibility layer that's able to pass the various function and argument struct pointers to a C++ type that can hold them. (I haven't dug into the details, it's just a general thought from the time I've spent in this code.)

Reply:
@dbaston Less ambitious attempt in https://github.com/OSGeo/gdal/pull/10964